addons/mail/static/src/models/notification_group_manager/notification_group_manager.js

```diff
@@ -48,8 +48,10 @@ function factory(dependencies) {
                 } else {
                     date = moment.max(group.date, notification.message.date);
                 }
+                const sequence = -Math.max(group.sequence, notification.message.id);
```

Comment:
I don't think this will work at all.

As soon as any negative value will be written, the next id which is positive will always be bigger no matter it's actual value.

(Also maybe not in this flux but we do have the potential for temporary messages with negative ids).

But overall this existing imperative algorithm is just not good. I think it would be time to make a proper compute for it (also for the other values).
So for that compute what needs to be done is to store all the related notifications on the group, and then compute in one go when the value for all notifications that are known.